processing/src/main/java/io/druid/segment/IndexMergerV9.java

```diff
@@ -536,73 +541,78 @@ private void makeInvertedIndexes(
+                  )
+              );
+            }
           }
-        }
 
-        MutableBitmap bitset = bitmapSerdeFactory.getBitmapFactory().makeEmptyMutableBitmap();
-        for (Integer row : CombiningIterable.createSplatted(
-            convertedInverteds,
-            Ordering.<Integer>natural().nullsFirst()
-        )) {
-          if (row != INVALID_ROW) {
-            bitset.add(row);
+          MutableBitmap bitset = bitmapSerdeFactory.getBitmapFactory().makeEmptyMutableBitmap();
+          for (Integer row : CombiningIterable.createSplatted(
+              convertedInverteds,
+              Ordering.<Integer>natural().nullsFirst()
+          )) {
+            if (row != INVALID_ROW) {
+              bitset.add(row);
+            }
           }
-        }
 
-        ImmutableBitmap bitmapToWrite = bitmapSerdeFactory.getBitmapFactory().makeImmutableBitmap(bitset);
-        if ((dictId == 0) && (Iterables.getFirst(dimVals, "") == null)) {
-          bitmapToWrite = nullRowBitmap.union(bitmapToWrite);
-        }
-        bitmapIndexWriters.get(dimIndex).write(bitmapToWrite);
-
-        if (spatialIndexWriter != null) {
-          String dimVal = dimVals.get(dictId);
-          if (dimVal != null) {
-            List<String> stringCoords = Lists.newArrayList(SPLITTER.split(dimVal));
-            float[] coords = new float[stringCoords.size()];
-            for (int j = 0; j < coords.length; j++) {
-              coords[j] = Float.valueOf(stringCoords.get(j));
+          ImmutableBitmap bitmapToWrite = bitmapSerdeFactory.getBitmapFactory().makeImmutableBitmap(bitset);
+          if ((dictId == 0) && (Iterables.getFirst(dimVals, "") == null)) {
+            bitmapToWrite = nullRowBitmap.union(bitmapToWrite);
+          }
+          bitmapIndexWriters.get(dimIndex).write(bitmapToWrite);
+
+          if (spatialIndexWriter != null) {
+            String dimVal = dimVals.get(dictId);
+            if (dimVal != null) {
+              List<String> stringCoords = Lists.newArrayList(SPLITTER.split(dimVal));
+              float[] coords = new float[stringCoords.size()];
+              for (int j = 0; j < coords.length; j++) {
+                coords[j] = Float.valueOf(stringCoords.get(j));
+              }
+              tree.insert(coords, bitset);
             }
-            tree.insert(coords, bitset);
           }
         }
+        if (spatialIndexWriter != null) {
+          spatialIndexWriter.write(ImmutableRTree.newImmutableFromMutable(tree));
+        }
+        log.info(
+            "Completed dim[%s] inverted with cardinality[%,d] in %,d millis.",
+            dimension,
+            dimVals.size(),
+            System.currentTimeMillis() - dimStartTime
+        );
       }
-      if (spatialIndexWriter != null) {
-        spatialIndexWriter.write(ImmutableRTree.newImmutableFromMutable(tree));
+      finally {
+        ByteBufferUtils.unmap(dimValsMapped);
```

Comment:
@drcrallen not 100% sure what do you mean, but since (as far as I understand) `Closer` is a try-with-resources emulation, with the only reason to use it since Java 7 of reducing code nesting, when multiple resources are declared in multiple locations, and there is only one resource here, simple try-with-resources should be ok